front/src/applications/operationalStudies/hooks/useLazyLoadTimatableItems.ts

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable no-restricted-syntax, no-await-in-loop */
+import { useEffect, useState, type Dispatch, type SetStateAction, useMemo } from 'react';
+
+import { osrdEditoastApi, type SimulationSummaryResult } from 'common/api/osrdEditoastApi';
+import type { TimetableItemWithDetails } from 'modules/trainschedule/components/Timetable/types';
+import type { TimetableItemId, TimetableItemWithTimetableId } from 'reducers/osrdconf/types';
+import { getBatchPackage } from 'utils/batch';
+import {
+  formatEditoastTrainIdToPacedTrainId,
+  formatEditoastTrainIdToTrainScheduleId,
+  formatPacedTrainIdToEditoastTrainId,
+  formatTrainScheduleIdToEditoastTrainId,
+  isTrainSchedule,
+} from 'utils/trainId';
+import { concatMap, mapBy } from 'utils/types';
+
+import formatTimetableItemSummaries from '../helpers/formatTimetableItemSummaries';
+
+const BATCH_SIZE = 10;
+
+type useLazyLoadTimetableItemsParams = {
+  infraId?: number;
+  electricalProfileSetId?: number;
+  timetableItemIdsToFetch?: TimetableItemId[];
+  timetableItems?: TimetableItemWithTimetableId[];
+  setTimetableItemIdsToFetch?: Dispatch<SetStateAction<TimetableItemId[] | undefined>>;
+  setTimetableItemIdsToProject?: Dispatch<SetStateAction<Set<TimetableItemId>>>;
+};
+
+/**
+ * This hook gradually fetches and projects items of the timetable.
+ *
+ * It first fetches the simulation of 10 items at a time, then projects them on the path.
+ * This optimizes the performance of the application and allow us to display the items as
+ * soon as they are ready.
+ */
+const useLazyLoadTimetableItems = ({
+  infraId,
+  electricalProfileSetId,
+  timetableItemIdsToFetch,
+  timetableItems,
+  setTimetableItemIdsToProject,
+}: useLazyLoadTimetableItemsParams) => {
+  const [timetableItemSummariesById, setTimetableItemSummariesById] = useState<
+    Map<TimetableItemId, TimetableItemWithDetails>
+  >(new Map());
+  const [allTimetableItemsLoaded, setAllTimetableItemsLoaded] = useState(false);
+
+  const [postTrainScheduleSimulationSummary] =
+    osrdEditoastApi.endpoints.postTrainScheduleSimulationSummary.useLazyQuery();
+
+  const [postPacedTrainSimulationSummary] =
+    osrdEditoastApi.endpoints.postPacedTrainSimulationSummary.useLazyQuery();
+
+  const { data: { results: rollingStocks } = { results: null } } =
+    osrdEditoastApi.endpoints.getLightRollingStock.useQuery({ pageSize: 1000 });
+
+  const timetalbeItemsById = useMemo(() => mapBy(timetableItems, 'id'), [timetableItems]);
+
+  // gradually fetch the simulation of the timetable items
+  useEffect(() => {
+    const getTimetableItemSummaries = async (
+      _infraId: number,
+      _timetableItemIdsToFetch: TimetableItemId[]
+    ) => {
+      setAllTimetableItemsLoaded(false);
+
+      for (let i = 0; i < _timetableItemIdsToFetch.length; i += BATCH_SIZE) {
```

Comment:
Hum do you think it would be better to fetch first all the paced Trains and then all the trainschedules (or the reverse ?) 
Otherwise, we might send twice more requests (if in each package we have both trainSchedules and pacedTrains, we might send each time a request for 5 trainSchedules and a request for 5 pacedTrains)